src/Workspaces/Core/Portable/CodeGeneration/CodeGenerationHelpers.cs

```diff
@@ -102,6 +103,42 @@ public static bool IsSpecialType(ITypeSymbol type, SpecialType specialType)
             return type != null && type.SpecialType == specialType;
         }
 
+        public static bool IsFlowAnalysisAttribute(AttributeData attribute)
```

Comment:
📝 We may need to avoid the allocations if this is to move forward, but that's a mechanical change that we can wait to do later.